src/ppx/register.ml

```diff
@@ -74,23 +74,22 @@ let switches = [
 
 let deprecated = Common.deprecated "bisect_ppx" [@coverage off]
 
-let switches =
-  switches
+let () = switches
   |> deprecated "-exclude"
   |> deprecated "-exclude-file"
   |> deprecated "-conditional"
   |> deprecated "-no-comment-parsing"
   |> Arg.align
-
+  |> List.iter (fun (key, spec, doc) -> Ppxlib.Driver.add_arg key spec ~doc)
 
 
 let () =
-  Migrate_parsetree.Driver.register
-    ~name:"bisect_ppx" ~args:switches ~position:100
-    Migrate_parsetree.Versions.ocaml_411 begin fun _config _cookies ->
-      match enabled () with
+  let impl =
+    match enabled () with
```

Comment:
BuckleScript instrumentation is still done unconditionally, unlike in `master`. This is probably because you have moved this `enabled ()` check outside of the suspension (line 90 in the old code) and out here, so that it runs immediately upon module initialization.

This is probably the same reason why the `dune-conditional` integration test is failing here: https://travis-ci.org/github/aantron/bisect_ppx/jobs/757439779#L611-L615. To run these yourself, run `make usage` in the project root directory.

This seems to be the last issue with the PR. I'll take care of the remaining two failures in Travis later. One is about updating some diffs for the self-instrumentation test, and the other looks like a difference (a bug fix) in `-dsource` output causing a spurious test failure on 4.12 (IIRC I reported the original output bug while writing these Bisect tests in the first place, causing harm to Bisect maintenance when it was fixed :P).

Reply:
Nice! That's what I was meaning before, but I didn't notice that on master that problem didn't exist since there `enabled ()` gets checked inside the callback. I've done the same here now. Thanks! Locally, `make usage` passes now, so I hope on the CI it will, too.